src/Filter/Filter.php

```diff
@@ -26,32 +26,40 @@ abstract class Filter implements FilterInterface
     protected $name;
 
     /**
-     * @var mixed|null
+     * @var mixed
      */
     protected $value;
 
     /**
-     * @var array
+     * @var array<string, mixed>
      */
     protected $options = [];
 
     /**
-     * @var string
+     * @var string|null
      */
     protected $condition;
 
-    public function initialize($name, array $options = []): void
+    public function initialize(string $name, array $options = []): void
     {
         $this->name = $name;
         $this->setOptions($options);
     }
 
-    public function getName()
+    public function getName(): string
     {
+        if (null === $this->name) {
+            throw new \RuntimeException(sprintf(
```

Comment:
If the should be fixed with a code change, then it should be a `LogicException`
```suggestion
            throw new \LogicException(sprintf(
```

Reply:
Indeed.

Dont know if the right way to think, but I generally have the rule:
Logic exception is something you should fix
Runtime exception is something you should catch

And I dont think a user should use this method inside a try catch.

Reply:
Fixed